### DISCUSSION TITLE
win aarch64 building is broken

### DISCUSSION THREAD

**Discussion by cstkingkey** [fltk-rs/fltk-rs#1395] [General]
win aarch64 building is broken

error[E0308]: mismatched types
   --> C:\Users\John\.cargo\registry\src\index.crates.io-6f17d22bba15001f\fltk-1.4.2\src\printer.rs:308:49
    |
308 |         unsafe { Fl_Printer_set_property_cancel(msg.into_raw()) }
    |                  ------------------------------ ^^^^^^^^^^^^^^ expected `*const u8`, found `*mut i8`
    |                  |
    |                  arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*mut i8`
note: function defined here
   --> C:\Users\John\.cargo\registry\src\index.crates.io-6f17d22bba15001f\fltk-sys-1.4.2\src\printer.rs:143:12
    |
143 |     pub fn Fl_Printer_set_property_cancel(msg: *const cty::c_char);
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

cty related.

**Comment by MoAlyousef** [comment]
Hi
Thanks for reporting this. Can you try with the current master branch?

**Comment by cstkingkey** [comment]
https://github.com/rust-lang/rust/blob/5133e154590991203c84702ef59482f729b19bbf/library/core/src/ffi/mod.rs#L104
under windows c_char is i8 . 

maybe cty need a fix？

**Reply by MoAlyousef** [reply]
Yeah it seems cty might benefit from a fix, I’m not sure if they would continue updating the crate now that core contains these type aliases.

**Reply by MoAlyousef** [reply]
I’ve removed the dependency on cty and tried a github action building for arm64 windows. It should build now.

**Comment by cstkingkey** [comment]
Now it works.
